### PR TITLE
Strongly typed core & geometries folders

### PR DESCRIPTION
**Pull request by dmk-rib**

### CODE REVIEW

**Comment by dmk-rib** [comment]
@wasabia This commit is the next portion of strongly typed classes.

File: lib/three3d/core/BufferAttribute.dart [hunk 1]
```diff
@@ -330,84 +330,97 @@ class BufferAttribute extends BaseBufferAttribute {
...
   }
 }
 
-class Int8BufferAttribute extends BufferAttribute {
-  String type = "Int8BufferAttribute";
-
-  Int8BufferAttribute(array, itemSize, [bool normalized = false])
-      : super(array, itemSize, normalized) {}
+class Int8BufferAttribute extends BufferAttribute<Int8List> {
+  Int8BufferAttribute(Int8List array, int itemSize, [bool normalized = false])
```
Comment (wasabia) [inline lib/three3d/core/BufferAttribute.dart:351]: Hi @dmk-rib 
BufferAttribute
array should be Dart list like Int8List or native array from flutter_gl Int8Array

if use Int8List need memory copy every frame in some cases.

or we should use native array from flutter_gl

or we support two? let array is dynamic?

Comment (wasabia) [reply lib/three3d/core/BufferAttribute.dart:351]: do you have any good idea?

Comment (dmk-rib) [reply lib/three3d/core/BufferAttribute.dart:351]: In my opinion, it is worth using the native array then. I can prepare a version with NativeArraies and send you for review, is it ok?

Comment (wasabia) [reply lib/three3d/core/BufferAttribute.dart:351]: ok

**Pull request merged**